### PR TITLE
Introducing an easier way to mount secrets and configmaps

### PR DESCRIPTION
**Pull request by bianchi2**
This PR introduces a new prop to values.yaml called additionalConfigMapsSecrets which is a list of configmaps and secrets with params. This is a more convenient way to mount configmaps and secrets vs existing functionality that makes it possible to pass additional volumeMounts and volumes separately. 

Example. Declaring the following configmaps and  a secret, each of which has **name** (existing secret or cm name), type (allowed values: configMap and secret), **mountPath** - i.e. destination directory in a container and **key** which is a key in secret/cm AND a destination file in the container):

```
additionalConfigMapsSecrets:
  - name: custom-config
    type: configMap
    key: log4j.properties
    mountPath:  /var/atlassian
  - name: custom-config
    type: configMap
    key: web.xml
    mountPath: /var/atlassian
  - name: keystore
    type: secret
    key: keystore.jks
    mountPath: /var/ssl
```
This results in the following volumeMounts:

```
volumeMounts:
  - name: local-home
    mountPath: "/var/atlassian/application-data/confluence"
  - name: local-home
    mountPath: "/opt/atlassian/confluence/logs"
    subPath: "logs"
  - name: shared-home
    mountPath: "/var/atlassian/application-data/shared-home"

  - name: custom-config-0
    mountPath: /var/atlassian/log4j.properties
    subPath: log4j.properties
  - name: custom-config-1
    mountPath: /var/atlassian/web.xml
    subPath: web.xml
  - name: keystore-2
    mountPath: /var/ssl/keystore.jks
    subPath: keystore.jks
```
It is possible to use one cm with multiple files in it (use various keys), and volumeMount and volume names will have cm/secret name + their index position, like custom-config-0, custom-config-1 or keystore-2. This is done to ask a customer just for 3 things - cm/secret name, key and destination in a container.

I have checked it with Bitbucket:

```
root@bibutcket-bitbucket-0:/var/atlassian/application-data/bitbucket# cat /var/atlassian/log4j.properties 
foo=bar
root@bibutcket-bitbucket-0:/var/atlassian/application-data/bitbucket# cat /var/atlassian/web.xml 
<xml></xml>
root@bibutcket-bitbucket-0:/var/atlassian/application-data/bitbucket# ls -ls /var/ssl/keystore.jks 
4 -rw-r--r--. 1 root bitbucket 1147 Apr  8 05:23 /var/ssl/keystore.jks
root@bibutcket-bitbucket-0:/var/atlassian/application-data/bitbucket# cat /var/ssl/keystore.jks 
-----BEGIN CERTIFICATE-----
MIIDITCCAgmgAwIBAgIRANSoSNVtKSl6xVK3cE9ZJYQzDQYJKoZIhvcNAQELBQAw
IDEeMBwGA1UEAxMVYXdzLWlhbS1hdXRoZW50aWNhdG9yMCAXDTIwMDcyNzE2Mzk1
OVoYDzIxMjAwNzAzMTYzOTU5WjAgMR4wHAYDVQQDExVhd3MtaWFtLWF1dGhlbnRp
Y2F0b3IwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQD3As+FPSBtDMDq
UvS2AiooNXpdkBCBCVrlqB7Hm4u2e3iY5rmaf7mh7B86WTd6WwyFhm2Cc4RZq6Ce
93Q2Pv9Ngi5v6Wyq0j76lIC7oJipp4TqsnnTV+bxFCnoKdYlPl7QysCDSVdjjNd6
dZlby1CiDNyOj0ZMS4NZchzyoyayZtZMJMvP0C2LEe/qNyxdn+lohdOE5M3K9rz0
MONDH2fKegZV2iYn69tBwLQStKmmeHQWvVTAr3AGDQo02duUXvb/AppGOEkXF8Mt
o/5igsK0dikR2RZL0l0r9DDVPUuvph9gAdoFCPpZogFqJqz0YF8Us7KbN5gbvW9T
ojvxYuWNAgMBAAGjVDBSMA4GA1UdDwEB/wQEAwICpDATBgNVHSUEDDAKBggrBgEF
BQcDATAPBgNVHRMBAf8EBTADAQH/MBoGA1UdEQQTMBGCCWxvY2FsaG9zdIcEfwAA
ATANBgkqhkiG9w0BAQsFAAOCAQEAAs4Y4xKye2G+zaijFNSNVuUOiSthVvK2AwRY
ovwNCf3NPQcEELTiT1kPvX9ZX4x8+FvMcZBIE7SOwhLKR5Z9a9qf0Wlbfj+mrVUb
jawwAC84tsbd+Z6QGGyYZczxv9xrftsgQw8HUR3Y2h4swP7Dv7QISzOmOpk6yMF0
MqAsJWl2L/FzNy3U9aCGc+cTu2huIjXGXJkE0xk7PhuVYKeB3ne+HbLWMxn/FrzT
7wA98GwA1VdkIv64+NVNbjHiiPPork4Fi9ZewABEAhYN5jeNymNKCtsS6gsyP46y
cRGRjawYCv6/Bwy9ptuZL3VUyiPQbmPB5SM2FwyLD51HsLJ8SQ==
-----END CERTIFICATE-----
```

### CODE REVIEW

File: src/main/charts/bitbucket/README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Kubernetes: `>=1.17.x-0`
 
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
+| additionalConfigMapsSecrets | list | `[]` | Additional existing ConfigMaps and Secrets not managed by Helm that should be mounted into server container configMap and secret are two available types (camelCase is important!) mountPath is a destination directory in a container and key is file name name references existing ConfigMap or secret name. VolumeMount and Volumes are added with this name + index position, for example custom-config-0, keystore-2 |
```
Comment (pbruski) [inline src/main/charts/bitbucket/README.md:21]: Hmmm, can we come up with a better name?
additionalConfigMapsAndSecrets ?
The name ephasises the source of data, while the user will likely be interested in the target, so maybe addtionalFiles ?

Comment (bianchi2) [reply src/main/charts/bitbucket/README.md:21]: yeah, not the best name indeed :) I am ok with additionalFiles. I just wanted the name to be self explanatory. I'll commit changes shortly.

Comment (bianchi2) [reply src/main/charts/bitbucket/README.md]: Renamed

Comment (pbruski) [reply src/main/charts/bitbucket/README.md]: Typo: addtionalFiles, `i` missing.

Comment (bianchi2) [reply src/main/charts/bitbucket/README.md]: thanks @pbruski . Fixed :)

**Review by pbruski — APPROVED** [review]

**Comment by dylanrathbone** [comment]
@bianchi2 for the sake of clarity, instead of introducing 1 new prop (`additionalConfigMapsSecrets`) for both maps and secrets, what do you think about introducing  2:

- `additionalConfigMaps` 
- `additionalSecrets`

This may be easier said than done, I realise that there may be some challenges in utilising both props where needed

**Review by dylanrathbone — APPROVED** [review]

**Pull request merged**